### ISSUE TITLE
v0.97b1 not a suitable build host for Larceny's before rev:5759

### ISSUE THREAD

**Issue by larceny-trac-import** [larcenists/larceny#612]
**Reported by: pnkfelix  on Wed Feb 25 17:52:19 2009**
The below text was originally written (stream of conciousness style) in the log for changeset:6090.

I'm filing this ticket so that I remember to double-check this problem in the future.  (Or at least check that a more recent development build of Larceny does not have a similar problem...)

---

FYI: there is definitely some problem when one uses v0.97b1 to host a
build of the `incrsumz/` branch (I just did it on artichoke on a
clean checkout of `incrsumz/`, which is what has delayed my
attempt to check this in.  I am not sure how the poblano build got
corrupted in a similar manner; I may have done a build on Feb 13th
using a private development build rather than the system installed
Larceny.  Or maybe the fact that the problem is exposed via v0.97b1
rather than v0.963 is a red herring.

But in any case, while in `branches/incrsumz/larceny_src` on
`artichoke.ccs.neu.edu`, with a clean checkout:

```
% larceny -- setup.sch -e "(setup-load-build 'sassy)"
Larceny v0.97b1 (beta test) (Aug 25 2008 12:37:27, precise:Linux:unified)
Loading Larceny compatibility package.
src/Compat/Larceny/compat2.sch
  ...
Compiling File (with Syntax) lib/SRFI/srfi-69.fasl
larceny.heap, built on Mon Aug 25 12:39:23 EDT 2008
> (exit)
% ./larceny.bin -stopcopy -- src/Build/iasn-larceny-heap.fasl
  ...
Welcome. Type (help) for help.
Install twobit's macro expander as the interpreter's ditto
Replace and populate the top-level environment
src/Lib/Common/toplevel.fasl
src/Lib/Arch/IAssassin/toplevel-target.fasl
done with load
WARNING from macro expander:
Redefining 
case-lambda


Error: fxlogand: 268435448 is not a fixnum.
Entering debugger; type "?" for help.
debug> b
=> compiled procedure debug/enter-debugger
   system continuation
   compiled procedure lgand
   compiled procedure #f
     ...
   compiled procedure #f
   compiled procedure sassy
     ...
debug> 
```

in other words, this is some sort of problem internal to the compiler;
from what I've been able to see, it seems to be some sort of weird
problem with the execution of the `fxlogand` primop...  other than
that, I do not know what's going on.  I cannot take the time to
investigate it more thoroughly at the moment, but one should be able
to do a checkout of the v0.97b1 code (on the `trunk/` from August
25th) and replicate the problem by attempting to build this code base
(on the `incrsumz/` branch revision 6090)

(Just saying "the trunk from August 25th" is more vague than I needed to be; I believe its revision 5783 of the trunk, in case that helps.)
